official/resnet/ctl/ctl_imagenet_main.py

```diff
@@ -215,70 +228,44 @@ def step_fn(inputs):
-    stats = {}
+    time_callback.on_train_begin()
     for epoch in range(train_epochs):
-      logging.info('Starting to run epoch: %s', epoch)
       train_iterator = iter(train_ds)
 
       step = 0
       total_loss = 0.0
-      batch_exp_per_sec = []
       for step in range(train_steps):
-        start_time = time.time()
-        learning_rate = compute_learning_rate(
-            epoch + 1 + (float(step) / steps_per_epoch))
-        optimizer.lr = learning_rate
-        start_time = time.time()
-        total_loss += train_step(next(train_iterator))
-        end_time = time.time()
-        elapsed_time = end_time - start_time
-        samples_per_sec = flags_obj.batch_size / elapsed_time
-        # We skip the first step for warmup purposes. We can
-        # add a flag for tuning this.
-        if step > 0:
-          batch_exp_per_sec.append(samples_per_sec)
+        training_accuracy.reset_states()
+        optimizer.lr = keras_common.learning_rate_schedule(
+            epoch, step, train_steps, flags_obj.batch_size)
 
+        time_callback.on_batch_begin(step+epoch*train_steps)
+        total_loss += train_step(next(train_iterator))
+        time_callback.on_batch_end(step+epoch*train_steps)
         step += 1
       train_loss = total_loss / step
-      # calculate average examples per second for a given epoch
-      epoch_exp_per_sec.append(np.mean(batch_exp_per_sec))
-      logging.info('Learning rate at epoch %s is %s',
-                   epoch, optimizer.lr.numpy())
-      logging.info('Training loss: %s, accuracy: %s%%',
-                   train_loss.numpy(),
-                   training_accuracy.result().numpy())
-      logging.info(
-          "Training Metric: {'epoch':%d, 'examples_per_second': %f}" %
-          (epoch, epoch_exp_per_sec[epoch]))
-
-      # Store the last train loss and accuracy calculated
-      stats['train_loss'] = train_loss.numpy()
-      stats['train_acc'] = training_accuracy.result().numpy()
-      training_accuracy.reset_states()
-
-      if (not flags_obj.skip_eval and
+
+      if (not flags_obj.skip_eval and epoch != 0 and
           epoch % flags_obj.epochs_between_eval == 0):
-        test_iterator = iter(test_ds)
-        for step in range(steps_per_eval):
-          test_step(next(test_iterator))
-        logging.info('Test loss: %s, accuracy: %s%%',
-                     test_loss.result().numpy(),
-                     test_accuracy.result().numpy())
-        stats['accuracy_top_1'] = test_accuracy.result().numpy()
-        stats['eval_loss'] = test_loss.result().numpy()
-        logging.info(
-            "Testing Metric: {'epoch':%d, 'test accuracy': %f}" %
-            (epoch, test_accuracy.result().numpy()))
         test_loss.reset_states()
         test_accuracy.reset_states()
 
-  stats['exp_per_second'] = np.mean(epoch_exp_per_sec)
-  return stats
+        test_iterator = iter(test_ds)
+        for step in range(steps_per_eval):
+          test_step(next(test_iterator))
+    
```

Comment:
Would be good to print some stuff every epoch/every few steps or so? Is anything being printed now at all?

Reply:
We log the time at the end of training so I wasn't sure if we should be logging anything. However this is not used for the perf number calculation(only for reporting test run time on the dashboard) so I added logging back.